### PR TITLE
Eos 15326 libfab

### PR DESCRIPTION
**Pull request by VinothVijayakumar-dev**
[EOS-15326] Validate libfabric code  using m0net ping ST
         - Added test script test-libfab.sh
         - Added libfab init and fini functions

### CODE REVIEW

File: net/libfab/libfab.c [hunk 1]
```diff
@@ -74,6 +74,26 @@ struct transfer_ma {
 	*/
 };
 
+/* libfab init and fini() : initialized in motr init */
+M0_INTERNAL int m0_net_libfab_init(void)
+{
+        int result = 0;
+     
+        /*  Uncomment it when all the changes are intergated 
+         *  m0_net_xprt_register(&m0_net_libfab_xprt);
```
Comment (upendrapatwardhan) [inline net/libfab/libfab.c:83]: Uncomment call for m0_net_xprt_register

File: net/libfab/libfab.c [hunk 2]
```diff
@@ -74,6 +74,26 @@ struct transfer_ma {
...
+        
+         *  m0_net_xprt_default_set(&m0_net_libfab_xprt);
+         */
+        return M0_RC(result);
+}
+
+M0_INTERNAL void m0_net_libfab_fini(void)
+{
+        /*  Uncomment it when all the changes are intergated 
+         *  m0_net_xprt_deregister(&m0_net_libfab_xprt);
```
Comment (upendrapatwardhan) [inline net/libfab/libfab.c:93]: Uncomment call for m0_net_xprt_deregister

File: net/libfab/libfab_internal.h [hunk 3]
```diff
@@ -22,8 +22,8 @@
 
 #pragma once
 
-#ifndef __MOTR_NET_LIBFAB_H__
-#define __MOTR_NET_LIBFAB_H__
+#ifndef __MOTR_NET_LIBFAB_INTERVAL_H__
```
Comment (upendrapatwardhan) [inline net/libfab/libfab_internal.h:25]: __MOTR_NET_LIBFAB_INTERNAL_H__

File: motr/init.c [hunk 4]
```diff
@@ -186,6 +189,7 @@ struct init_fini_call subsystem[] = {
 	{ &m0_addb2_net_module_init, &m0_addb2_net_module_fini, "addb2-net" },
 #ifndef __KERNEL__
 	{ &m0_net_sock_mod_init, &m0_net_sock_mod_fini, "net/sock" },
+        { &m0_net_libfab_init,   &m0_net_libfab_fini,   "net/libfab" },
```
Comment (huanghua78) [inline motr/init.c:192]: alignment.

File: net/libfab/libfab_internal.h (hunk 3 above)
Comment (huanghua78) [reply net/libfab/libfab_internal.h:25]: MOTR_NET_LIBFAB_LIBFAB_INTERNAL_H

File: net/libfab/libfab.h [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+#pragma once
+
+#ifndef __MOTR_NET_LIBFAB_H__
```
Comment (huanghua78) [inline net/libfab/libfab.h:25]: __MOTR_NET_LIBFAB_LIBFAB_H__

File: net/test/test-libfab.sh.in [hunk 6]
```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
```
Comment (huanghua78) [inline net/test/test-libfab.sh.in:1]: What is the difference of this test script and "net/test/test-user.sh.in"?

I see most of them are identical. 
So, can we use the same script?
If needed, we can use some arguments to differentiate them.

Comment (VinothVijayakumar-dev) [reply net/test/test-libfab.sh.in]: Initially the arguments of m0nettest are wrong and the end points of lnet and libfab are different.
so created a new test script to run libfab test.

--added an option to chose lnet or libfab in test-user.sh and modified options of lnet as properly.

**Review by upendrapatwardhan — APPROVED** [review]

File: net/test/test-user.sh.in [hunk 7]
```diff
@@ -29,6 +29,16 @@ if [ $1 != "console" -a $1 != "server" -a $1 != "client" ]; then
 	exit
 fi
 
+echo "--------------------Test-user-script------------------------"
+echo -n "Enter (1)Lnet (2)libFab :"
```
Comment (jamohite) [inline net/test/test-user.sh.in:33]: can we add socket transport ?

Comment (VinothVijayakumar-dev) [reply net/test/test-user.sh.in:33]: this is based on endpoints changes. For Sock need to know the endpoints required  and I think we can run the same option of libfab, and the actual transport choice is inside m0_net.

**Review by huanghua78 — APPROVED** [review]

**Comment by huanghua78** [comment]
This is reviewed. Is it ready to be megerd?

**Pull request merged**